src/cart/crt_rpc.c

```diff
@@ -1689,6 +1689,12 @@ crt_rpc_common_hdlr(struct crt_rpc_priv *rpc_priv)
 					crt_ctx->cc_rpc_cb_arg);
 	} else {
 		rpc_priv->crp_opc_info->coi_rpc_cb(&rpc_priv->crp_pub);
+		/*
+		 * Correspond to crt_rpc_handler_common -> crt_rpc_priv_init's
+		 * set refcount as 1.
+		 */
+		if (rpc_priv->crp_srv)
+			RPC_DECREF(rpc_priv);
```

Comment:
i dont think this is correct/needed. we call rpc handler locally, but we still wait for all children of collective rpc to finish before we destroy this rpc

Reply:
We should decrement reference here to make the counter the same as in customized path.

Reply:
Ahh. Previously it was decremented in crt_rpc_handler_common(). I think this is incorrect. The logic should be the same as possible for both paths customized and not.

Reply:
hmm i see. i am guessing you are referring to the path of first if statement in crt_handle_rpc():
1627         if (rpc_priv->crp_srv || (rpc_priv->crp_coll && !rpc_priv->crp_srv))
1628                 RPC_DECREF(rpc_priv);

Reply:
yes